packages/profile-sync-controller/src/controllers/user-storage/services.ts

```diff
@@ -268,6 +268,49 @@ export async function deleteUserStorage(
   }
 }
 
+/**
+ * User Storage Service - Delete multiple storage entries for one specific feature.
+ * You cannot use this method to delete multiple features at once.
+ *
+ * @param data - data to delete, in the form of an array entryKey[]
+ * @param opts - storage options
+ */
+export async function batchDeleteUserStorage(
+  data: string[],
+  opts: UserStorageBatchUpsertOptions,
+): Promise<void> {
+  if (!data.length) {
+    return;
+  }
+
+  const { bearerToken, path, storageKey } = opts;
+
+  const encryptedData: string[] = [];
+
+  for (const d of data) {
+    encryptedData.push(createSHA256Hash(d + storageKey));
+  }
+
+  const url = new URL(`${USER_STORAGE_ENDPOINT}/${path}`);
+
+  console.log('body', `batch_delete: ${JSON.stringify(encryptedData)}`);
+  console.log('url', url.toString());
+
+  const res = await fetch(url.toString(), {
+    method: 'PUT',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${bearerToken}`,
+    },
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    body: JSON.stringify({ batch_delete: encryptedData }),
```

Comment:
hehehe, so this is how we will support batch delete, via a PUT 😄 

Fair, nicely done!